packages/rsocket-core/src/RSocketClient.js

```diff
@@ -183,13 +186,19 @@ class RSocketClientSocket<D, M> implements ReactiveSocket<D, M> {
       lifetime,
       metadata,
       metadataMimeType,
+      payload,
     } = config.setup;
+    const data =
+      payload === undefined
+        ? undefined
+        : JsonSerializer.serialize(payload);
```

Comment:
This should really be using `serializers` above, but it looks like `metadata` is just passed through as is so we're already kinda wrong here. For that reason I think we should do exactly the same here for payload (just make it a string and pass through?).